packages/hoppscotch-data/src/environment.ts

```diff
@@ -59,36 +65,71 @@ export const parseBodyEnvVariables = (
 
 export function parseTemplateStringE(
   str: string,
-  variables: Environment["variables"]
+  variables: Environment["variables"],
+  myVariables?: Variables | undefined
 ) {
+
   if (!variables || !str) {
     return E.right(str)
   }
 
   let result = str
-  let depth = 0
+  let depthEnv = 0
+  let depthVar = 0
 
-  while (result.match(REGEX_ENV_VAR) != null && depth <= ENV_MAX_EXPAND_LIMIT) {
+  while (result.match(REGEX_ENV_VAR) != null && depthEnv <= ENV_MAX_EXPAND_LIMIT) {
     result = decodeURI(encodeURI(result)).replace(
       REGEX_ENV_VAR,
       (_, p1) => variables.find((x) => x.key === p1)?.value || ""
     )
-    depth++
+    depthEnv++
   }
 
-  return depth > ENV_MAX_EXPAND_LIMIT
-    ? E.left(ENV_EXPAND_LOOP)
-    : E.right(result)
+  if (myVariables) {
+    while (result.match(REGEX_MY_VAR) != null && depthVar <= ENV_MAX_EXPAND_LIMIT) {
```

Comment:
We have to consider cases where path variables expand to environment variables, we have to resolve then those as well.